templates/deployment.yaml

```
@@ -63,7 +68,7 @@ spec:
   volumeClaimTemplates: {{ toYaml . | nindent 6 }}
   {{- end }}
   {{- end }}
-  {{- if eq ( default $serviceDefinition.type "Deployment" ) "Job" }}
+  {{- if eq ( $serviceDefinition.type | default "Deployment" ) "Job" }}
```

This mix it difficult to read => 
```suggestion
  {{- if eq ( default "Deployment" $serviceDefinition.type ) "Job" }}
```

---

your suggestion is putting back what I had to remove to make it work. When you use the synthax ` default "Deployment" $serviceDefinition.type` then `$serviceDefinition.type` ALWAYS `Deployment` thus all the tests about Statufulset and jobs do not work